bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/internal/handler/PigpioRemoteHandler.java

```diff
@@ -57,60 +63,368 @@ public class PigpioRemoteHandler extends BaseThingHandler {
      */
     public PigpioRemoteHandler(Thing thing) {
         super(thing);
+        logger.debug("gpio : ctor");
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        ChannelHandler channelHandler = channelHandlers.get(channelUID);
-        if (channelHandler != null) {
-            channelHandler.handleCommand(command);
+        try {
+            synchronized (this.connectionLock) {
+                ChannelHandler channelHandler = channelHandlers.get(channelUID);
+
+                if (channelHandler == null || !(ThingStatus.ONLINE.equals(thing.getStatus()))) {
+                    // We raced with connectPollWorker and lost
+                    return;
+                }
+
+                if (channelHandler instanceof PigpioDigitalInputHandler) {
+                    PigpioDigitalInputHandler inputHandler = (PigpioDigitalInputHandler) channelHandler;
+
+                    try {
+                        inputHandler.handleCommand(command);
+                    } catch (PigpioException pe) {
+                        logger.warn("Input command exception on channel {} {}", channelUID, pe.toString());
+                        if (pe.getErrorCode() == -99999999) {
+                            runDisconnectActions();
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                    pe.getLocalizedMessage());
+                        }
+                    }
+                } else if (channelHandler instanceof PigpioDigitalOutputHandler) {
```

Comment:
```suggestion
                } else if (channelHandler instanceof PigpioDigitalOutputHandler outputHandler) {
```
java17 instanceof pattern matching